OutOfSchool/OutOfSchool.WebApi/Controllers/V2/CompetitiveEventController.cs

```diff
@@ -249,10 +514,155 @@ private CompetitiveEventResponseDto CreateUpdateResponse(CompetitiveEventResultD
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest(ModelState);
+        }
+
+        try
+        {
+            await competitiveEventDraftService.Reject(id, competitiveEventDraftRejection.RejectionMessage);
+            return Ok();
+        }
+        catch (EntityDeletedConflictException ex)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+
+        }
+        catch (EntityModifiedConflictException ex)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// Approves the competitive event draft by a user with permission to moderate drafts.
+    /// </summary>
+    /// <param name="id">Key in the table.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    [HasPermission(Permissions.CompetitiveEventApprove)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPut("/api/v{version:apiVersion}/competitions-drafts/{id}/approve")]
+    public async Task<IActionResult> Approve(Guid id)
+    {
+        try
+        {
+            await competitiveEventDraftService.Approve(id);
+            return Ok();
+        }
+        catch (EntityDeletedConflictException ex)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+
+        }
+        catch (EntityModifiedConflictException ex)
+        {
+            return StatusCode(StatusCodes.Status500InternalServerError, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// Deletes the cover image from the specified competitive event draft by a user with permission to moderate drafts.
+    /// </summary>
+    /// <param name="draftId">The ID of the draft to update.</param>
+    /// <returns>Returns <see cref="CompetitiveEventDraftResponseDto"/> with the cover image removed if successful.</returns>
+    /// <response code="200">Cover image deleted successfully.</response>
+    /// <response code="400">The draft does not have a cover image.</response>
+    /// <response code="401">The user is not authenticated.</response>
+    /// <response code="403">The user does not have permission to perform this action.</response>
+    /// <response code="404">The specified draft was not found.</response>
+    /// <response code="409">The draft is not editable.</response>
+    /// <response code="500">An unexpected error occurred.</response>
+    [HttpDelete("/api/v{version:apiVersion}/competitions-drafts/{draftId}/cover-image")]
+    [HasPermission(Permissions.CompetitiveEventEdit)]
```

Comment:
I agree with rabbit. Only people with approve rights should be able to edit drafts other than people who belong to provider. And we have `CompetitiveEventApprove = 154` permission